xdsl/irdl/constraints.py

```diff
@@ -677,17 +677,21 @@ def get_variable_extractors(
         """
         return {}
 
-    def can_infer(self, var_constraint_names: Set[str]) -> bool:
+    def can_infer(self, var_constraint_names: Set[str], length_known: bool) -> bool:
```

Comment:
At the call site, the `True` or `False` don't communicate much in terms of what's going on, how do you feel about making the argument mandatory?
```suggestion
    def can_infer(self, var_constraint_names: Set[str], *, length_known: bool) -> bool:
```